### PR TITLE
feat(controller): add ps:restart command

### LINKED ISSUE #467: Should be able to restart container processes
Heroku has `ps:restart` and Deis needs something similar to bounce databases or other processes that may become stuck.

### PR DESCRIPTION
**Pull request by bacongobbler**
closes #467

### CODE REVIEW

**Comment by nathansamson** [comment]
by the looks of it these are not rolling restarts. Would it be possible to make that an option to minimize downtime?

**Comment by bacongobbler** [comment]
In a future PR, sure. I'm sure there's going to be many different implementations where users are going to want different restart policies, so this is just the simplest solution for now and we can work on a proposal for rolling restarts later.

**Comment by bacongobbler** [comment]
I still need to implement `ps:restart web.2` as per https://devcenter.heroku.com/articles/dynos#cli-commands-for-dyno-management. Commit coming up

**Comment by bacongobbler** [comment]
done. Once jenkins agrees with me this is ready for code review

File: controller/scheduler/states.py [hunk 1]
```diff
@@ -1,9 +1,14 @@
 import enum
 
 
-class TransitionNotAllowed(Exception):
+class TransitionError(Exception):
```
Comment (mboersma) [inline controller/scheduler/states.py]: Is this more generic name better for this exception class? Just curious.

**Comment by mboersma** [comment]
Code LGTM, nice work! It looks like there is good test coverage, but let me know if you think this needs manual testing as well.

File: controller/scheduler/states.py (hunk 1 above)
Comment (bacongobbler) [inline controller/scheduler/states.py]: I saw documentation in python https://docs.python.org/2/tutorial/errors.html#user-defined-exceptions and thought it was better

**Comment by bacongobbler** [comment]
Whoops, forgot to bump API docs as well

File: controller/scheduler/states.py (hunk 1 above)
Comment (mboersma) [inline controller/scheduler/states.py]: > Most exceptions are defined with names that end in “Error,” similar to the naming of the standard exceptions.

Works for me. +1.

**Comment by bacongobbler** [comment]
@mboersma I think the unit/integration tests speak for themself :)

**Comment by carmstrong** [comment]
:ship:

**Comment by bacongobbler** [comment]
Jenkins derped at the last minute, but this did pass all tests: https://ci.deis.io/job/test-ec2/722/console

Good to :shipit:

**Comment by carmstrong** [comment]
> Good to ?

I'd say so.

**Pull request merged**